airbyte-integrations/connectors/source-instagram/source_instagram/client/api.py

```diff
@@ -272,13 +286,10 @@ def list(self, fields: Sequence[str] = None) -> Iterator[dict]:
                 )
                 yield clear_url(record_data)
 
-    @backoff_policy
     def _get_media(self, instagram_user: IGUser, params: dict = None, fields: Sequence[str] = None) -> Iterator[Any]:
-        """
-        This is necessary because the functions that call this endpoint return
-        a generator, whose calls need decorated with a backoff.
-        """
-        return instagram_user.get_media(params=params, fields=fields)
+        media = self.get_instance_cursor(instagram_user, "get_media", params=params, fields=fields)
+        for media_obj in self.pagination(media):
+            yield media_obj
```

Comment:
`_get_media` and `_get_stories` looks identical except "get_*" attribute name string.
We can consider moving `get_instance_cursor` into `pagination` and extend `pagination` parameters, e. g.:
```
for media_obj in self.pagination("get_media"):
    yield media_obj
```
Or we can consider having wrapper function for the 3 lines above.